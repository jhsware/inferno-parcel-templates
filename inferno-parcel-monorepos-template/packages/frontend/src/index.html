<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8"/>
    <title>Inferno Monorepo Template</title>
    <script type="module" src="index.ts"></script>
  </head>
  <body>
    <div id="app"></div>
  </body>
</html>