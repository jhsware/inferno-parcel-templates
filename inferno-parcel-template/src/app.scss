article {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  min-height: 30vh;
}