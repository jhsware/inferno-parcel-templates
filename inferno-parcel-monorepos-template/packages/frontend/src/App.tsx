import { Button } from 'components';
import './app.scss'

/**
 * The Application
 */
type TApp = {
  title: string,
  children: any
}
export function App({ title, children }: TApp) {
  return (
    <article>
      <h1>{title}</h1>
      {children}
    </article>
  )
};

export function appFactory () {

  return (
      <App title="Inferno Monorepo Template">
        <Button onClick={(e) => alert("This works!")}>Click me!</Button>
      </App>
  )
}
