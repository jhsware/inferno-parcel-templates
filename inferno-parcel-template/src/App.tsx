import { Button } from './components/Button';
import './app.scss'

/**
 * The Application
 */
type TApp = {
  title: string,
  children: any
}
export function App({ title, children }: TApp) {
  return (
    <article>
      <h1>{title}</h1>
      {children}
    </article>
  )
};

export function appFactory () {

  return (
      <App title="Inferno Template">
        <Button onClick={(e) => alert("This works!")}>Click me!</Button>
      </App>
  )
}
