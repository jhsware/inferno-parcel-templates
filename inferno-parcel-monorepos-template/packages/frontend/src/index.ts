import { render } from 'inferno';
import { appFactory } from './App';

render(appFactory(), document.getElementById('app'))
