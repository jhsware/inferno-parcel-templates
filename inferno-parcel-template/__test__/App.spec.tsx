import { render } from 'inferno';
import { describe, expect, it, beforeEach } from "@jest/globals";

import { App } from "../src/App";

describe('App', () => {
  let node;
  beforeEach(() => {
    node = document.createElement('div');
  })

  it('can be rendered', () => {
    render(<App title="Test">test</App>, node);
    expect(node).toMatchSnapshot();
  });
});