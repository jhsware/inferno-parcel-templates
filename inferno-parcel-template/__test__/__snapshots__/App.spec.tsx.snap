// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button can be rendered 1`] = `
<div>
  <article>
    <h1>
      Test
    </h1>
    test
  </article>
</div>
`;
